src/frontend/js/data/fetchApi.ts

```diff
@@ -0,0 +1,68 @@
+import { appFrontendContext } from './frontEndData';
+import {
+  Actions,
+  User,
+  RoutesApiModerator,
+} from './../components/DashboardModerators/common';
+import { handle } from '../utils/errors/handle';
+
+/**
+ * Request API to fetch list data
+ */
+export const fetchListUsers = async (type: RoutesApiModerator) => {
+  let response: Response;
+  try {
+    response = await fetch(`/api/v1/` + type, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+  } catch (error) {
+    return handle(error);
+  }
+  if (!response.ok) {
+    return handle(
+      new Error(`Failed to load list student : ${type} ${response.status}.`),
+    );
+  }
+  try {
+    const responseData = await response.json();
+    return responseData;
+  } catch (error) {
+    handle(new Error(`Failed to decode JSON in list ${type} ` + error));
+  }
+};
+
+/**
+ * Request API to add group moderator for this student
+ */
+export const updateStudentToModerator = async (user: User, action: Actions) => {
+  let response: Response;
+  try {
+    response = await fetch(`/api/v1/students/update/`, {
+      headers: {
+        'Content-Type': 'application/json',
+        'X-CSRFTOKEN': appFrontendContext.csrftoken,
+      },
+      body: JSON.stringify({
+        user_id: user.id,
+        action,
+      }),
+      method: 'POST',
+    });
+  } catch (error) {
+    return handle(error);
+  }
+  if (!response.ok) {
+    return handle(new Error(`Failed to update student : ${response.status}.`));
+  }
+  let responseData: User;
+  try {
+    responseData = await response.json();
```

Comment:
```suggestion
    return await response.json();
```

Reply:
I haven't done this one because if I don't return User type, it upsets code.